python/graphstorm/config/argument.py

```diff
@@ -203,6 +203,113 @@ def override_arguments(self, cmd_args):
                 setattr(self, f"_{arg_key}", arg_val)
                 print(f"Overriding Argument: {arg_key}")
 
+    def verify_arguments(self, is_train):
+        """ Verify the correctness of arguments.
+
+        Parameters
+        ----------
+        is_train : bool
+            Whether this is for training.
+        """
+        # Trigger the checks in the arguments.
+        _ = self.save_perf_results_path
+        _ = self.profile_path
+        _ = self.graph_name
+        _ = self.backend
+        _ = self.ip_config
+        _ = self.part_config
+        _ = self.node_id_mapping_file
+        _ = self.edge_id_mapping_file
+        _ = self.verbose
+
+        # Data
+        _ = self.node_feat_name
+        _ = self.decoder_edge_feat
+
+        # Evaluation
+        _ = self.eval_fanout
+        _ = self.use_mini_batch_infer
+        _ = self.eval_batch_size
+        _ = self.eval_frequency
+        _ = self.no_validation
+        _ = self.save_prediction_path
+        _ = self.eval_negative_sampler
+        _ = self.eval_etype
+        _ = self.eval_metric
+        _ = self.num_negative_edges_eval
+
+        # Model training.
+        if is_train:
+            _ = self.batch_size
+            _ = self.fanout
+            _ = self.lm_train_nodes
+            _ = self.lm_tune_lr
+            _ = self.lr
+            _ = self.sparse_optimizer_lr
+            _ = self.num_epochs
+            _ = self.save_model_path
+            _ = self.save_model_frequency
+            _ = self.topk_model_to_save
+            _ = self.early_stop_burnin_rounds
+            _ = self.early_stop_rounds
+            _ = self.early_stop_strategy
+            _ = self.use_early_stop
+            _ = self.wd_l2norm
+            _ = self.train_negative_sampler
+            _ = self.train_etype
+            _ = self.remove_target_edge_type
+
+        # LM module
+        _ = self.lm_infer_batch_size
```

Comment:
We only need to check lm related module when 
`if self.node_lm_configs is not None`